multibody/kinematic/kinematic_evaluator_set.cc

```diff
@@ -117,22 +152,54 @@ template <typename T>
 int KinematicEvaluatorSet<T>::add_evaluator(KinematicEvaluator<T>* e) {
   // Compare plants for equality by reference
   DRAKE_DEMAND(&plant_ == &e->plant());
-  
+
   evaluators_.push_back(e);
   return evaluators_.size() - 1;
 }
 
 template <typename T>
 std::vector<int> KinematicEvaluatorSet<T>::FindUnion(
-    KinematicEvaluatorSet<T> other) {
-  std::vector<int> union_index;
+    KinematicEvaluatorSet<T> other) const {
+  std::vector<int> union_indices;
+  for (int i = 0; i < other.num_evaluators(); i++) {
+    if (std::find(evaluators_.begin(), evaluators_.end(),
+                  &other.get_evaluator(i)) != evaluators_.end()) {
+      union_indices.push_back(i);
+    }
+  }
+  return union_indices;
+}
+
+template <typename T>
+std::vector<int> KinematicEvaluatorSet<T>::FindFullIndicesUnion(
+    KinematicEvaluatorSet<T> other) const {
+  std::vector<int> row_indices;
+  for (int i = 0; i < other.num_evaluators(); i++) {
+    if (std::find(evaluators_.begin(), evaluators_.end(),
+                  &other.get_evaluator(i)) != evaluators_.end()) {
+      int start_index = other.evaluator_full_start(i);
+      for (int j = 0; j < other.get_evaluator(i).num_full(); j++) {
+        row_indices.push_back(start_index + j);
+      }
+    }
+  }
+  return row_indices;
+}
+
+template <typename T>
+std::vector<int> KinematicEvaluatorSet<T>::FindActiveIndicesUnion(
+    KinematicEvaluatorSet<T> other) const {
+  std::vector<int> row_indices;
   for (int i = 0; i < other.num_evaluators(); i++) {
     if (std::find(evaluators_.begin(), evaluators_.end(),
-        other.get_evaluator(i)) != evaluators_.end()) {
-      union_index.push_back(i);
+                  &other.get_evaluator(i)) != evaluators_.end()) {
+      int start_index = other.evaluator_active_start(i);
+      for (int j = 0; j < other.get_evaluator(i).num_active(); j++) {
+        row_indices.push_back(start_index + j);
```

Comment:
Returning indices of the full constraint might be better, and I think it's consistent to the rest of the APIs. 

The code probably looks like this: 
```
      int start_index = other.evaluator_full_start(i);
      for (int j = 0; j < other.get_evaluator(i).num_active(); j++) {
        row_indices.push_back(start_index + other.get_evaluator(i).active_inds().at(j));
      }
```